src/Action/EditAction.php

```diff
@@ -118,7 +120,13 @@ protected function _put($id = null)
             $this->saveOptions()
         );
 
-        $this->_trigger('beforeSave', $subject);
+        $event = $this->_trigger('beforeSave', $subject);
+        if ($event->isStopped()) {
+            $subject->set(['success' => false]);
+
+            return $this->_stopped($subject);
```

Comment:
`_stopped()` redirects to index action. Shouldn't a failed save render the form instead with submitted data as it's done when `Model.beforeSave` fails?

Reply:
So should `_stopped()` have a second arg that allows disabling the redirect?

Reply:
I guess so.